odl/solvers/functional/functional.py

```diff
@@ -929,8 +948,10 @@ def __init__(self, left, right):
             raise TypeError('`func` {} is not a `Functional` instance'
                             ''.format(right))
 
+        # HELP: What should happen if left.range != right.range?
```

Comment:
IMO either nothing, or raise early whatever error adding the convex conjugates would raise. I tend towards not doing anything and letting `convex_conj` fail, at least if the error message is good enough to tell users what exactly went wrong.

Another way to play this would be to strictly require `RealNumbers` as `range` for both, since from the mathematical definition it wouldn't make sense otherwise. However, as in the convex conjugate there's probably an adapted definition for complex-valued functionals, so we probably shouldn't close the door here.